internal/pkg/utils/random.go

```diff
@@ -0,0 +1,32 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package utils
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// MakeRandomGetter returns a function that can be used to randomly access the given store.
+func MakeRandomGetter[V, T any](ln func(V) int, getFn func(V, int) T) func(v V) (T, bool) {
```

Comment:
do we need tests for this?

Reply:
On further thinking about this, I decided to remove this thing entirely. It's a leftover from the time before I moved to using slices for upstreams.